task/fbc-validation/0.1/fbc-validation.yaml

```diff
@@ -162,16 +165,42 @@ spec:
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
+
         if ! ${OPM_BINARY} validate ."${conffolder}"; then
           echo "!FAILURE! - opm validate check failed."
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
+
         if ! ${OPM_BINARY} render ."${conffolder}" | jq -en 'reduce (inputs | select(.schema == "olm.package")) as $obj (0; .+1) == 1'; then
           echo "!FAILURE! - More than one olm.packages is not permitted in a FBC fragment."
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
+
+        OCP_BUNDLE_METADATA_THRESHOLD_MAJOR=4
+        OCP_BUNDLE_METADATA_THRESHOLD_MINOR=17
+
+        if [[ "${OCP_VER_MAJOR}" -ge "${OCP_BUNDLE_METADATA_THRESHOLD_MAJOR}" ]] && [[ "${OCP_VER_MINOR}" -ge "${OCP_BUNDLE_METADATA_THRESHOLD_MINOR}" ]]; then
+          #
+          # above the version threshold for mandatory `olm.csv.metadata` bundle metadata format, so presence of `olm.bundle.object` is an error
+          #
+          if ! ${OPM_BINARY} render ."${conffolder}" | jq -en 'reduce( inputs | select(.schema == "olm.bundle" and .properties[].type == "olm.bundle.object")) as $_ (0;.+1) == 0'; then
```

Comment:
How expensive is the opm render call? 
We can maybe consider rendering once and saving to a JSON file so we can run these kinds of checks on that file instead.

Reply:
They're linear based on the number of provided channels/bundles in the fragment.  I'd like to render once and re-examine, but was not sure where to drop the interim render output.  I'll look for existing convention somewhere, unless you can direct me.